app/helpers/search_helper.rb

```diff
@@ -11,6 +11,11 @@ def clear_filters_path
 
   def filter_link(name, value, count, html_options = {}, &block)
     parameters = search_and_facet_params
+
+    # TODO: is there a better way?
+    if name == 'language'
+      title = render_language_name(value)
+    end
     title ||= (html_options.delete(:title) || truncate(value.to_s, length: 50))
```

Comment:
Not currently, but I would move this logic into a method that takes the facet name and value and handles the special case for `language`, otherwise falls back to `truncate(value.to_s, length: 50)` as below.